### DISCUSSION TITLE
Compiling mesmerizer

### DISCUSSION THREAD

**Discussion by jansaell** [PlummersSoftwareLLC/NightDriverStrip#417] [General]
HI there!

I have finally had the time to start working on the mesmerizer board.

And thanks David for the shipment.

I changed the platform.in line to say mesmerizer

[platformio]
default_envs = mesmerizer

And when i compile it i get some strange errors:
Compiling .pio/build/mesmerizer_lolin/src/network.cpp.o
In file included from .pio/libdeps/mesmerizer_lolin/FastLED/src/platforms/esp/32/fastled_esp32.h:6,
                 from .pio/libdeps/mesmerizer_lolin/FastLED/src/platforms.h:42,
                 from .pio/libdeps/mesmerizer_lolin/FastLED/src/FastLED.h:62,
                 from include/globals.h:107,
                 from src/main.cpp:160:
.pio/libdeps/mesmerizer_lolin/FastLED/src/platforms/esp/32/fastspi_esp32.h:2:17: note: #pragma message: ESP32 Hardware SPI support added
 #pragma message "ESP32 Hardware SPI support added"
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from include/effectdependencies.h:82,
                 from src/effects.cpp:32:
include/effects/matrix/PatternWeather.h: In member function 'virtual void PatternWeather::Draw()':
include/effects/matrix/PatternWeather.h:368:38: error: no matching function for call to 'GFXBase::fillScreen(CRGB)'
         g()->fillScreen(CRGB(0, 0, 0));
                                      ^
In file included from include/gfxbase.h:68,
                 from include/globals.h:1684,
                 from src/effects.cpp:30:
.pio/libdeps/mesmerizer_lolin/Adafruit GFX Library/Adafruit_GFX.h:62:16: note: candidate: 'virtual void Adafruit_GFX::fillScreen(uint16_t)'
   virtual void fillScreen(uint16_t color);
                ^~~~~~~~~~
.pio/libdeps/mesmerizer_lolin/Adafruit GFX Library/Adafruit_GFX.h:62:16: note:   no known conversion for argument 1 from 'CRGB' to 'uint16_t' {aka 'short unsigned int'}
In file included from include/effectdependencies.h:82,
                 from src/effectfactories.cpp:31:
include/effects/matrix/PatternWeather.h: In member function 'virtual void PatternWeather::Draw()':
include/effects/matrix/PatternWeather.h:368:38: error: no matching function for call to 'GFXBase::fillScreen(CRGB)'
         g()->fillScreen(CRGB(0, 0, 0));
                                      ^
In file included from include/gfxbase.h:68,
                 from include/globals.h:1684,
                 from include/effectdependencies.h:34,
                 from src/effectfactories.cpp:31:
.pio/libdeps/mesmerizer_lolin/Adafruit GFX Library/Adafruit_GFX.h:62:16: note: candidate: 'virtual void Adafruit_GFX::fillScreen(uint16_t)'
   virtual void fillScreen(uint16_t color);
                ^~~~~~~~~~
.pio/libdeps/mesmerizer_lolin/Adafruit GFX Library/Adafruit_GFX.h:62:16: note:   no known conversion for argument 1 from 'CRGB' to 'uint16_t' {aka 'short unsigned int'}
Compiling .pio/build/mesmerizer_lolin/src/screen.cpp.o
*** [.pio/build/mesmerizer_lolin/src/effects.cpp.o] Error 1
*** [.pio/build/mesmerizer_lolin/src/effectfactories.cpp.o] Error 1

What am i doing wrong?

**Comment by davepl** [comment]
I think you spelled mesmerizer wrong, that’s all!

**Comment by jansaell** [comment]
Sorry - got a exra n when patins in - and no - its copied from the env section.
I have editid the post now.

**Comment by davepl** [comment]
If you don’t have any pending work in that folder, I’d wipe it and do a fresh git clone.

In the unlikely event that doesn’t reset everything, try 

pio package update

From a terminal within VS.

- Dave

**Comment by rbergen** [comment]
@jansaell As far as I can tell, there are two problems:

1. The logging shows a build of the mesmerizer_lolin environment, not "regular" mesmerizer. This may be because of the exact command you use to start the build, but that I can't see. In any case, you are building the wrong variety of mesmerizer.
2. You're building an old version of the NightDriverStrip codebase. At some point a bug did exist that yielded the error messages you're seeing, but it got fixed quite a while back. Also, when I look at my clone, the code at the line numbers mentioned in your logging don't match with the code the logging shows. I therefore think it's necessary for you to update your local copy of the code.

**Comment by jansaell** [comment]
thanks - should slap myself for not thinking about that. Now it works.

**Reply by robertlipe** [reply]
Please consider marking this issue as 'fixed'. Thank you.
